### PR TITLE
Deprecated tool in favor of components and services used as tools

### LINKED ISSUE #196: Deprecate tool in favor of component or service

### PR DESCRIPTION
**Pull request by stevespringett**
Deprecated tool in favor of components and services used as tools. Added both current 1.5 and deprecated examples of each.

Closes #196

### CODE REVIEW

File: schema/bom-1.5.schema.json [hunk 1]
```diff
@@ -137,11 +137,39 @@
...
+          "oneOf": [
+            {
+              "type": "object",
+              "title": "Creation Tools",
+              "description": "The tool(s) used in the creation of the BOM.",
+              "additionalProperties": false,
+              "properties": {
+                "components": {
+                  "type": "array",
+                  "additionalItems": false,
```
Comment (prabhu) [inline schema/bom-1.5.schema.json:149]: What would be nice is to ask for relevant configuration, parameters, and environment variables used by the component or the service to generate the SBoM. Inspired by commandLine and arguments in sarif.

https://docs.oasis-open.org/sarif/sarif/v2.1.0/os/sarif-v2.1.0-os.html#_Toc34317570

File: schema/bom-1.5.schema.json [hunk 2]
```diff
@@ -137,11 +137,39 @@
...
+                  "items": {"$ref": "#/definitions/component"},
+                  "uniqueItems": true,
+                  "title": "Components",
+                  "description": "A list of software and hardware components used as tools"
+                },
+                "services": {
+                  "type": "array",
+                  "additionalItems": false,
```
Comment (prabhu) [inline schema/bom-1.5.schema.json:157]: Can the service list the underlying components used to generate the SBoM?

File: schema/bom-1.5.schema.json (hunk 1 above)
Comment (stevespringett) [reply schema/bom-1.5.schema.json:149]: That's all part of formulation #31

File: schema/bom-1.5.schema.json (hunk 2 above)
Comment (stevespringett) [reply schema/bom-1.5.schema.json:157]: The service can have an external reference to its corresponding SBOM

File: tools/src/test/resources/1.5/valid-vulnerability-1.5.json [hunk 3]
```diff
@@ -80,19 +80,22 @@
...
+            "name": "Snyk CLI (Linux)",
+            "version": "1.729.0",
+            "hashes": [
+              {
+                "alg": "SHA-256",
+                "content": "2eaf8c62831a1658c95d41fdc683cd177c147733c64a93e59cb2362829e45b7d"
+              }
+            ]
+          }
+        ]
```
Comment (jkowalleck) [inline tools/src/test/resources/1.5/valid-vulnerability-1.5.json:97]: could you add `services`?
so we would have a complete example.

File: tools/src/test/resources/1.5/valid-vulnerability-1.5.xml [hunk 4]
```diff
@@ -80,14 +80,16 @@
...
+                <components>
+                    <component type="application">
+                        <group>Snyk</group>
+                        <name>Snyk CLI (Linux)</name>
+                        <version>1.729.0</version>
+                        <hashes>
+                            <hash alg="SHA-256">2eaf8c62831a1658c95d41fdc683cd177c147733c64a93e59cb2362829e45b7d</hash>
+                        </hashes>
+                    </component>
+                </components>
```
Comment (jkowalleck) [inline tools/src/test/resources/1.5/valid-vulnerability-1.5.xml:92]: could you also add `<services>`? 
So we would have a complete example.

File: schema/bom-1.5.schema.json [hunk 5]
```diff
@@ -183,7 +211,7 @@
     "tool": {
       "type": "object",
       "title": "Tool",
-      "description": "Information about the automated or manual tool used",
+      "description": "[Deprecated] - DO NOT USE. This will be removed in a future version. This will be removed in a future version. Use component or service instead. Information about the automated or manual tool used",
```
[the remark this answers is not in the document]
Comment (stevespringett) [reply schema/bom-1.5.schema.json:214]: Adding an unknown property to a schema may make the resulting schema invalid to various parsers, depending on the "strictness" of the parser.

File: tools/src/test/resources/1.5/valid-vulnerability-1.5.json (hunk 3 above)
Comment (stevespringett) [reply tools/src/test/resources/1.5/valid-vulnerability-1.5.json:108]: Good catch. Added in https://github.com/CycloneDX/specification/pull/198/commits/bba0493f9c44faa67e436eee1fe0685c152ad6d7

File: tools/src/test/resources/1.5/valid-vulnerability-1.5.xml (hunk 4 above)
Comment (stevespringett) [reply tools/src/test/resources/1.5/valid-vulnerability-1.5.xml:92]: Good catch. Added in https://github.com/CycloneDX/specification/pull/198/commits/bba0493f9c44faa67e436eee1fe0685c152ad6d7

**Review by jkowalleck — APPROVED** [review]

**Pull request merged**